core/src/main/java/org/apache/iceberg/io/DeleteSchemaUtil.java

```diff
@@ -43,4 +43,15 @@ public static Schema pathPosSchema() {
   public static Schema posDeleteSchema(Schema rowSchema) {
     return rowSchema == null ? pathPosSchema() : pathPosSchema(rowSchema);
   }
+
+  public static Schema posDeleteReadSchema(Schema rowSchema) {
```

Comment:
Somehow after the rebase this is needed.  

Note that all the readers used now seem to no longer even read the 'row' field https://github.com/apache/iceberg/blob/main/data/src/main/java/org/apache/iceberg/data/BaseDeleteLoader.java#L70. 

But as this is a rewrite, I am taking the strategy of RewritePositionDelete and actually reading this field, but as optional to avoid the assert error if it is not found.  https://github.com/apache/iceberg/blob/main/core/src/main/java/org/apache/iceberg/PositionDeletesTable.java#L118. (the reader there is derived from schema of metadata table PositionDeletesTable).

Reply:
This might not directly related to this PR, but seems like the column `row` should NOT be marked as `required` anywhere.